test/e2e/main_test.go

```diff
@@ -108,24 +111,97 @@ func TestNodeStartup(t *testing.T) {
-		panic("Labels haven't been propagated to pods")
+
+	// Checking labels propagation to pods
+	// from Infinispan CR to pods
+	if pod.Labels["my-pod-label"] != ispn.Labels["my-pod-label"] {
+		panic("Infinispan CR labels haven't been propagated to pods")
 	}
 
+	// from operator environment
+	if tconst.RunLocalOperator == "TRUE" {
+		// running locally, labels are hardcoded and set by the testsuite
+		if pod.Labels["operator-pod-label"] != "operator-pod-value" ||
+			ispn.Labels["operator-pod-label"] != "operator-pod-value" {
+			panic("Infinispan CR labels haven't been propagated to pods")
+		}
+	} else {
+		// Get the operator namespace from the env if it's different
+		// from the testsuite one
+		operatorNS := tconst.OperatorNamespace
+		if operatorNS == "" {
+			operatorNS = spec.Namespace
+		}
+		// operator deployed on cluster, labels are set by the deployment
+		if !areOperatorLabelsPropagated(operatorNS, ispnv1.OperatorPodTargetLabelsEnvVarName, pod.Labels) {
+			panic("Operator labels haven't been propagated to pods")
+		}
+	}
+
+	// Checking labels propagation to all the services
 	svcList := &corev1.ServiceList{}
 	testutil.ExpectNoError(kubernetes.Kubernetes.ResourcesList(ispn.ObjectMeta.Name, ispn.ObjectMeta.Namespace, "", svcList))
 	if len(svcList.Items) == 0 {
 		panic("No services found for cluster")
 	}
 	for _, svc := range svcList.Items {
-		if svc.Labels["my-svc-label"] != ispn.Labels["my-svc-label"] ||
-			svc.Labels["operator-svc-label"] != "operator-svc-value" ||
-			ispn.Labels["operator-svc-label"] != "operator-svc-value" {
-			panic("Labels haven't been propagated to services")
+		// from Infinispan CR to service
+		if svc.Labels["my-svc-label"] != ispn.Labels["my-svc-label"] {
+			panic("Infinispan CR labels haven't been propagated to services")
+		}
+
+		// from operator environment
+		if tconst.RunLocalOperator == "TRUE" {
+			// running locally, labels are hardcoded and set by the testsuite
+			if svc.Labels["operator-svc-label"] != "operator-svc-value" ||
+				ispn.Labels["operator-svc-label"] != "operator-svc-value" {
+				panic("Labels haven't been propagated to services")
+			}
+		} else {
+			// Get the operator namespace from the env if it's different
+			// from the testsuite one
+			operatorNS := tconst.OperatorNamespace
+			if operatorNS == "" {
+				operatorNS = spec.Namespace
+			}
+			// operator deployed on cluster, labels are set by the deployment
+			if !areOperatorLabelsPropagated(operatorNS, ispnv1.OperatorTargetLabelsEnvVarName, svc.Labels) {
+				panic("Operator labels haven't been propagated to services")
+			}
+		}
+	}
+}
+
+// areOperatorLabelsPropagated helper function that read the labels from the infinispan operator pod
+// and match them with the labels map provided by the caller
+func areOperatorLabelsPropagated(namespace, varName string, labels map[string]string) bool {
+	labelSelector, err := labutil.Parse("name=infinispan-operator")
```

Comment:
This label is "infinispan-operator-alm-owned" when deployed by OLM

Reply:
Also shouldn't the label state `olm` instead of `alm`?

Reply:
I've aligned this branch with the labels settings from master branch